cmd/config/internal/commands/cmdcreatesetter.go

```diff
@@ -138,6 +148,99 @@ func (r *CreateSetterRunner) preRunE(c *cobra.Command, args []string) error {
 	return nil
 }
 
+func (r *CreateSetterRunner) processSchema() error {
```

Comment:
I feel this should be baked into the api layer in kyaml so that it can be reusable from elsewhere. Any specific reason to have this logic in cmd/config ?

Reply:
I have moved the validation of the types in the schema into the `SetterCreator`. Some of the other checks need to remain here since we need to consolidate the type value from the schema and the `--type` flag before we make the check to verify that the `--field` flag is set if the type is `array`. 
I'll do a follow up PR to try to clean some of this up. We should probably remove the `Type` field from the `SetterCreator` and any logic that requires the type should find it in the schema. This might allow us to move additional checks from the command into the `SetterCreator`.